api/envoy/extensions/transport_sockets/proxy_protocol/v3/upstream_proxy_protocol.proto

```diff
@@ -24,4 +24,20 @@ message ProxyProtocolUpstreamTransport {
 
   // The underlying transport socket being wrapped.
   config.core.v3.TransportSocket transport_socket = 2 [(validate.rules).message = {required: true}];
+
+  // Control the connection pool settings for PROXY protocol socket.
+  // [#not-implemented-hide:]
+  ConnectionPoolSettings connection_pool_settings = 3;
+}
+
+// ConnectionPool settings for PROXY protocol socket.
+message ConnectionPoolSettings {
```

Comment:
I don't think "connection pool" is the right term here? These are socket settings? Or just ProxyProtocolSettings? Can these settings just be inlined?

/wait

Reply:
These are impacting envoy conn pool setting for proxy_procotol transport socket.

https://www.envoyproxy.io/docs/envoy/latest/intro/arch_overview/upstream/connection_pooling#number-of-connection-pools

I put here because the `ConnectionPoolSettings` is in the uptream_proxy_protocol package.
Do you mean inline in `ProxyProtocolUpstreamTransport`? 
Sure

Reply:
I think I would just inline the settings. The the user the fact that they impact connection pool sharing is an implementation details. I would just document as appropriate what each setting does.

Reply:
Thank you! Does the new proto align with your opinion?

Reply:
Yes this seems fine. I think you can just do the implementation as part of this PR.

/wait

Reply:
@mattklein123 Can you merge this first? If this api is merged, it unblocks me from starting to sync the API at my side